addons/account/models/account_move_line.py

```diff
@@ -2664,6 +2666,18 @@ def _prepare_analytic_distribution_line(self, distribution, account_id, distribu
             'category': 'invoice' if self.move_id.is_sale_document() else 'vendor_bill' if self.move_id.is_purchase_document() else 'other',
         }
 
+    def _round_analytic_distribution_line(self, analytic_lines):
+        """round the analytic lines amount, and cancel the rounding error"""
+        if not analytic_lines:
+            return
+
+        rounding_error = 0
+        for line in analytic_lines:
+            rounded_amount = self.currency_id.round(line['amount'])
+            rounding_error += rounded_amount - line['amount']
+            line['amount'] = rounded_amount
+        analytic_lines[-1]['amount'] -= self.currency_id.round(rounding_error)
```

Comment:
round *after* the operation
`analytic_lines[-1]['amount'] = self.currency_id.round(analytic_lines[-1]['amount'] - rounding_error)`